cvcreator/cvcreate.py

```diff
@@ -87,31 +88,31 @@ def main():
             content.update(config)
 
             template = src.get_template()
-            
+
             if not args.projects:
                 content.pop("Projects", None)
             else:
                 proj = content.pop("Projects", {})
+                proj_keys = tuple_of_strings(args.projects, proj)
                 content["Projects"] = {}
-                for i, pn in enumerate(sorted(proj)):
-                    n = int(pn[1:])
-                    assert n < MAX_NUM_CONST, "Does not support cases for indices larger than MAX_NUM_CONST"
-                    pi = "A%d" % i
-                    if n in args.projects:
-                        content["Projects"][pi] = proj.pop(pn)
-            
+                for i, key in enumerate(proj_keys):
+                    try:
+                        content["Projects"][i] = proj[key]
+                    except KeyError:
+                        raise KeyError(f"Key {key} not found in Projects")
```

Comment:
This type of re-raising is a bit unecesary as the raise condition you are looking for is much easier to get from a simple if-test:
```suggestion
                    if key not in proj:
                        raise KeyError(f"Key {key} not found in Projects")
                    content["Projects"][i] = proj[key]
```